### PR TITLE
feat: DEPR EdxRestApiClient

### PR DESCRIPTION
**Pull request by Yagnesh1998**
**Description:**

As Per this DEPR: https://github.com/openedx/public-engineering/issues/218

This ticket is to clean-up some remaining usages, and ultimately to remove the deprecated EdxRestApiClient so the code no longer lives in the platform.

Move EdxRestApiClient into a location near where it is used in edx-platform and rename it to DeprecatedRestApiClient

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Yagnesh1998! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by dianakhuang** [comment]
Another note: you will want to bump the version number for this library (in the __version__ file) to a new major version (so 6.0.0) and add a note to the README that there is a breaking change here.

**Comment by e0d** [comment]
Hi @Yagnesh1998 looks like there are some CI failures, can you have a look?

**Comment by Yagnesh1998** [comment]
@mphilbrick211lo  two workflows awaiting approval.

**Comment by mphilbrick211** [comment]
Hi @Yagnesh1998! I'm seeing that the workflows are enabled. Would you mind looking into the failing checks?

**Comment by Yagnesh1998** [comment]
@mphilbrick211 After adding a commit on this repo it always gives me 2 workflows waiting for approval. How can I do it myself?

**Comment by mphilbrick211** [comment]
@Yagnesh1998 - it looks like you should be able to rebase and have the tests run again. Let me know if that doesn't work.

**Comment by Yagnesh1998** [comment]
@mphilbrick211 doesn't work.

**Comment by Yagnesh1998** [comment]
@mphilbrick211 doesn't work.

File: edx_rest_api_client/tests/test_client.py
Comment (dianakhuang) [inline edx_rest_api_client/tests/test_client.py:146]: This causes this test to fail, because the `self.assertRaises` expected an error to happen in `EdxRestApiClient`.

File: edx_rest_api_client/tests/test_client.py
Comment (dianakhuang) [inline edx_rest_api_client/tests/test_client.py:84]: This fails because the api will only be called if `EdxRestApiClient.get_and_cache_jwt_oauth_access_token` gets called.

**Comment by Yagnesh1998** [comment]
@mphilbrick211 can u please approve so the test is starting to run?

**Comment by mphilbrick211** [comment]
@Yagnesh1998 - looks like all the tests ran and are passing.

@openedx/arch-bom - this is ready for review!

File: edx_rest_api_client/client.py
Comment (robrap) [inline edx_rest_api_client/client.py:9]: - Hopefully `slumber` can be fully removed from this library, including the requirements files. That means removing from the `*.in` requirement files, and then running `make upgrade` in the top-level directory to generate the updated `*.txt` requirement files. 
- It looks like slumber is also used in an exceptions file: https://github.com/search?q=repo%3Aopenedx%2Fedx-rest-api-client%20slumber&type=code
- You still need to do a major upgrade of the version to 6.0.0 here: https://github.com/openedx/edx-rest-api-client/blob/6bbb88e1325d3adc7efe9015bee004411b18780e/edx_rest_api_client/__version__.py#L1, and update the CHANGELOG.rst. See https://open-edx-proposals.readthedocs.io/en/latest/best-practices/oep-0051-bp-conventional-commits.html for handling backward incompatible changes like this one.

Comment (Yagnesh1998) [reply edx_rest_api_client/client.py:9]: @robrap Yes I need to do a version change and then add something to the CHANGELOG indicating that we are removing the old client and that this is a breaking change:

Comment (robrap) [reply edx_rest_api_client/client.py:9]: @Yagnesh1998: A reminder to update CHANGELOG.rst. See https://open-edx-proposals.readthedocs.io/en/latest/best-practices/oep-0051-bp-conventional-commits.html for handling backward incompatible changes like this one.

**Comment by robrap** [comment]
@dianakhuang: Do we want to add a local version to edx-platform, which will also require adding slumber to edx-platform directly, or is it possible to repeat the fixes that were done in the first place to retire the client?

File: edx_rest_api_client/tests/test_client.py
Comment (dianakhuang) [inline edx_rest_api_client/tests/test_client.py:129]: I think we can just delete this test.

File: edx_rest_api_client/tests/test_client.py
Comment (dianakhuang) [inline edx_rest_api_client/tests/test_client.py:147]: I think we can delete this test, too.

File: edx_rest_api_client/tests/test_client.py
Comment (dianakhuang) [inline edx_rest_api_client/tests/test_client.py:186]: And this one.

**Review by dianakhuang — COMMENTED** [review]
Could you also update CHANGELOG.rst for version 6.0.0 mentioning that we are removing `EdxRestApiClient` and that it's a breaking change?

**Comment by mphilbrick211** [comment]
@dianakhuang - would you mind enabling tests on this again? Thanks!

**Comment by mphilbrick211** [comment]
@robrap - is this something you can review / merge?

**Comment by dianakhuang** [comment]
@Yagnesh1998 could you rebase and resolve conflicts? I can re-review it once that's done.

**Comment by mphilbrick211** [comment]
Hi @Yagnesh1998 - checking in on this. Are you planning to pursue this pull request? If so, there's branch conflicts that need to be resolved before the pull request can be merged. Thanks!

**Comment by Yagnesh1998** [comment]
I will do that currently I am on holiday but I will finish that soon

**Comment by mphilbrick211** [comment]
Hi @Yagnesh1998 - friendly check in on this :)

**Comment by mphilbrick211** [comment]
Closing for now. Can reopen in the future if needed.

**Comment by openedx-webhooks** [comment]
@Yagnesh1998 Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-rest-api-client&entry.867055334=https://github.com/openedx/edx-rest-api-client/pull/303&entry.1484655318=https://github.com/Yagnesh1998&entry.752974735=2023-11-06+15:43&entry.1917517419=2024-08-22+21:02&entry.2133058324=No) so we can improve your experience in the future.

**Comment by feanil** [comment]
Gonna re-open and see if DEPR WG can get it over the line.

**Comment by feanil** [comment]
@robrap @dianakhuang this is good to go now I think, also, while doing this I noticed other maintenance that needs to be done so I made a PR for that here: https://github.com/openedx/edx-rest-api-client/pull/362

**Review by dianakhuang — APPROVED** [review]

**Pull request merged**
